src/system_database.ts

```diff
@@ -977,40 +984,73 @@ export class PostgresSystemDatabase implements SystemDatabase {
       query = query.limit(input.limit);
     }
     const rows = await query.select();
-    const workflows = rows.map((row) => { return {
-      workflowID: row.workflow_uuid,
-      queueName: row.queue_name,
-      createdAt: row.created_at_epoch_ms,
-      startedAt: row.started_at_epoch_ms,
-      completedAt: row.completed_at_epoch_ms,
-    }});
+    const workflows = rows.map((row) => {
+      return {
+        workflowID: row.workflow_uuid,
+        executorID: row.executor_id,
+        queueName: row.queue_name,
+        createdAt: row.created_at_epoch_ms,
+        startedAt: row.started_at_epoch_ms,
+        completedAt: row.completed_at_epoch_ms,
+      };
+    });
     return { workflows };
   }
 
   async enqueueWorkflow(workflowId: string, queue: WorkflowQueue): Promise<void> {
-    const _res = await this.pool.query<workflow_queue>(`
+    await this.pool.query<workflow_queue>(
+      `
       INSERT INTO ${DBOSExecutor.systemDBSchemaName}.workflow_queue (workflow_uuid, queue_name)
       VALUES ($1, $2)
       ON CONFLICT (workflow_uuid)
       DO NOTHING;
-    `, [workflowId, queue.name]);
+    `,
+      [workflowId, queue.name]
+    );
+  }
+
+  async reEnqueueWorkflow(workflowId: string): Promise<void> {
+    const client: PoolClient = await this.pool.connect();
+    await client.query("BEGIN ISOLATION LEVEL READ COMMITTED");
+    await client.query<workflow_queue>(
+      `
+      UPDATE ${DBOSExecutor.systemDBSchemaName}.workflow_queue
+      SET started_at_epoch_ms = NULL, executor_id = NULL
+      WHERE workflow_uuid = $1;
+    `,
+      [workflowId]
+    );
+    await client.query<workflow_status>(
+      `
+      UPDATE ${DBOSExecutor.systemDBSchemaName}.workflow_status
+      SET status = $2
+      WHERE workflow_uuid = $1;
```

Comment:
I think we should also set `executor_id = NULL` here. I don't think it has a functional impact, but it might be useful for debugging / observability.